.github/workflows/smoke-test.yaml

```diff
@@ -77,17 +76,22 @@ jobs:
             ${{ runner.os }}-gradle-
 
       - name: Build application
-        run: ${{ env.GRADLE_EXEC }} build
+        run: ${{ env.GRADLE_EXEC }} clean :server:build -x test
 
-      - name: Run application in background, capture logs in app.log
-        run: |
-          ${{ env.GRADLE_EXEC }} :server:run 2> server/src/test/resources/app.log < /dev/null &
-          echo "Application started with PID $APP_PID"
-          sleep 10
+      - name: Build Smoke Tests
+        run: ${{ env.GRADLE_EXEC }} buildSmokeTests
 
-      - name: Print App Logs
-        run: cat server/src/test/resources/app.log
+      # FIXME for POC, version and exit code are static, they need to be dynamic, see docker script
+      - name: Create & Start Container
+        working-directory: server/docker/
+        run: |
+          docker buildx build --load -t "block-node-server:0.1.0-SNAPSHOT" --build-context distributions=../build/distributions --build-arg VERSION="0.1.0-SNAPSHOT" . || exit "123"
+          docker compose -p block-node up -d
 
       - name: Smoke Test
         working-directory: server/src/test/resources/
-        run: ./smoke-test.sh app.log
+        run: ./smoke-test.sh
+
+      # TODO Do we need to stop this container? In theory when the job runs, everything gets scrapped
```

Comment:
I think we should stop the container and also check that the exit code is 0. I've seen a number of defects in the past which surface when shutting down.

Reply:
Noted.